doc/developer/design/20210311_prometheus_source.md

```diff
@@ -0,0 +1,144 @@
+# Summary
+
+This design doc introduces a way to import data from prometheus metrics endpoints into materialized views that can be inspected with SQL commands.
```

Comment:
This is about implementing a generic Prometheus metrics scraper in Materialize, correct? This is not about exposing internal system tables as Prometheus metrics to be scraped, correct?

Perhaps we should call this `Prometheus metrics source / scraper`?

Reply:
Yep, that's correct. Renaming sounds good (I thought it was pretty much implicit in the nature of the source, but good to make explicit!)

Reply:
I think I misunderstood our conversation, I thought that we were discussing just scraping our internal metrics (via registry::gather, aka your second alternative), without hitting any external endpoints. That seems like it would satisfy the requirements of getting _our_ metrics data into system tables without adding a lot of other complexity.

New prometheus source types are, I think, currently just skunkworks projects (which I would be interested in chatting about!) or deprioritized for a quarter or two.